src/components/data/datagrid/datagrid.tsx

```diff
@@ -703,11 +700,14 @@ export const DataGridHeading: React.FC<DataGridHeadingProps> = ({
   // Debounce filter
   useEffect(() => {
     const handler = () => {
-      onPageChange?.(1);
+      // No filter state.
+      if (filterState === undefined) {
+        return;
+      }
       onFilter(filterState || {});
     };
-    onFilterTimeoutRef.current && clearTimeout(onFilterTimeoutRef.current);
-    setTimeout(handler, 100);
+    clearTimeout(onFilterTimeoutRef.current);
```

Comment:
Is this ok in the first render when `onFilterTimeoutRef.current` is `null`?

Reply:
Should be ok:

```
> clearTimeout(null)
< undefined
> clearTimeout(undefined)
< undefined
```

Can add a guard just to be save